### PR TITLE
wip(proto): throw errors when backwards incompatible changes are added

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**
- Build a cue representation of the current proto file
- Compare it to the previous definitions to ensure that only backwards-compatible changes were introduced.

**Motivation for the change:**
Playing with cuelang

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: cue.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+func main() {
+	file, err := protobuf.Extract("pkg/api/registry.proto", nil, &protobuf.Config{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	b, _ := format.Node(file)
+	ioutil.WriteFile("out.cue", b, 0644)
```
Comment (ecordell) [inline cue.go:19]: convert the current proto file into cue and save it

File: cue.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	previnst, err := r.Compile("out_prev.cue", nil)
+	if err != nil {
+		log.Fatal("error compiling prev: ", err)
+	}
+	inst, err := r.Compile("out.cue", nil)
+	if err != nil {
+		log.Fatal("error compiling current: ", err)
+	}
+	var errs []error
+	it, err := previnst.Value().Fields(cue.Definitions(true))
```
Comment (ecordell) [inline cue.go:31]: find all definitions in the previous schema

File: cue.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	for it.Next() {
+		if !it.IsDefinition() {
+			continue
+		}
+		def, _ := it.Value().Label()
+		curr := inst.Value().LookupDef(def)
+		if err:= curr.Subsume(it.Value()); err != nil {
```
Comment (ecordell) [inline cue.go:41]: find the definition with the same name in the new schema and ensure that it "subsumes" the previous one (i.e. is backwards compatible)

**Comment by njhale** [comment]
closing due to staleness.

**Pull request closed without merging**